apps/src/javaide/JavaIdeView.jsx

```diff
@@ -1,22 +1,132 @@
 import React from 'react';
-import AceEditor from 'react-ace';
-import 'ace-builds/src-noconflict/mode-java';
-import 'ace-builds/src-noconflict/theme-monokai';
+import JavaConsole from './JavaConsole';
+import {connect} from 'react-redux';
+import JavaEditor from './JavaEditor';
+import PaneHeader, {PaneSection} from '@cdo/apps/templates/PaneHeader';
+import {appendOutputLog} from './redux';
+import PropTypes from 'prop-types';
 
-export default class JavaIdeView extends React.Component {
-  onChange = newValue => {
-    console.log('change', newValue);
+const style = {
+  instructionsAndPreview: {
+    width: '40%',
+    display: 'block'
+  },
+  editorAndConsole: {
+    width: '60%',
+    display: 'block'
+  },
+  instructions: {
+    height: '25%',
+    marginLeft: 15,
+    marginRight: 15,
+    backgroundColor: 'white',
+    color: 'black'
+  },
+  preview: {
+    margin: 15,
+    backgroundColor: 'gray',
+    height: '200px'
+  },
+  javaIde: {
+    display: 'flex',
+    backgroundColor: '#1b1c17',
+    margin: 15
+  },
+  consoleAndButtons: {
+    marginTop: 15,
+    display: 'flex'
+  },
+  consoleStyle: {
+    flexGrow: 1
+  },
+  buttons: {
+    marginRight: 15,
+    height: 75,
+    backgroundColor: '#272822',
+    display: 'flex',
+    justifyContent: 'center',
+    alignItems: 'center'
+  },
+  singleButton: {
+    margin: 15,
+    textAlign: 'center'
+  }
+};
+
+class JavaIdeView extends React.Component {
+  static propTypes = {
+    // populated by redux
+    editorText: PropTypes.string,
```

Comment:
Why does JavaIdeView need `editorText`? i.e. could it be used by JavaEditor only?

Reply:
good question--for now it doesn't actually need it, but if we keep the run button within this component (or put it anywhere other than the JavaEditor component) we will need access to the `editorText` to actually run the program. I can remove it for now since it's not needed, but I think it should stay in redux.

Reply:
Ahh ok. Makes sense. Whatever you decide is good with me!